var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -64,6 +64,11 @@ def install(self, spec, prefix):
                 'CXXFLAGS={0}'.format(self.compiler.pic_flag),
                 'FCFLAGS={0}'.format(self.compiler.pic_flag)]
 
+        # H5public.h includes mpi.h directly
+        if 'hdf5+mpi' in spec:
+            config_args.append(
+                'CPPFLAGS=-I{0}'.format(self.spec['mpi'].prefix.include))
```

Comment:
I think this will get you by the configuration failure but when Silo goes to build any of its executables (tools or tests), that will fail due to also needing LDFLAGS. So, I think you also need to add something like so...

```
LDFLAGS=-L{0} -lmpi'.(format(self.spec['mpi'].prefix.lib)
```

Reply:
I think the upshot from this is that Silo's configure could be a bit smarter about whether its been asked to use serial or parallel install of HDF5.

Reply:
It works even without `LDFLAGS`.

Reply:
Hmm...are you building Silo tools and/or tests? I can't think their links will complete using an MPI-enabled HDF5 without causing, somehow, -lmpi to appear on the linker command-line. Are you using MPI compiler wrappers by chance?

Reply:
I was trying to use `C{C,XX}=mpic{,xx}`, but that led to some linking error.

Anyhow, I think that is really not an silo issue, but an spack CC issue, which assumes that external package follow a standard layout, which isn't the case on Debian/Ubuntu.

Reply:
Using `spec["mpi"].headers.directories` (as defined in #7267) should fix it.